src/Compilers/CSharp/Test/Semantic/Semantics/SpanStackSafetyTests.cs

```diff
@@ -1008,19 +1009,208 @@ public static async Task<int> M1()
+                );
+        }
+
+        [Fact, WorkItem(62747, "https://github.com/dotnet/roslyn/issues/62747")]
+        public void AsyncLocals_Deconstruction()
+        {
+            var src = @"
+using System;
+using System.Threading.Tasks;
+
+public class Program
+{
+    public async Task M()
+    {
+        (Span<int> s1, Span<int> s2) = new Program(); // 1, 2
+        var (s3, s4) = new Program(); // 3, 4
+
+        await Task.Yield();
+        return;
+    }
+
+    void Deconstruct(out Span<int> s1, out Span<int> s2) => throw null;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(
+                // (9,10): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         (Span<int> s1, Span<int> s2) = new Program(); // 1, 2
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "Span<int>").WithArguments("System.Span<int>").WithLocation(9, 10),
+                // (9,24): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         (Span<int> s1, Span<int> s2) = new Program(); // 1, 2
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "Span<int>").WithArguments("System.Span<int>").WithLocation(9, 24),
+                // (10,14): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         var (s3, s4) = new Program(); // 3, 4
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "s3").WithArguments("System.Span<int>").WithLocation(10, 14),
+                // (10,18): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         var (s3, s4) = new Program(); // 3, 4
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "s4").WithArguments("System.Span<int>").WithLocation(10, 18)
+                );
+        }
+
+        [Fact, WorkItem(62747, "https://github.com/dotnet/roslyn/issues/62747")]
+        public void AsyncLocals_Using()
+        {
+            var src = @"
+using System.Threading.Tasks;
+
+public class Program
+{
+    public async Task M()
+    {
+        using (default(RS)) { } // 1
+        using (var s1 = default(RS)) { } // 2
+        using (RS s2 = default(RS)) { } // 3
+        using RS s3 = default(RS); // 4
+
+        await Task.Yield();
+        return;
+    }
+}
+
+public ref struct RS
+{
+    public void Dispose() { }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(
+                // (8,16): error CS4012: Parameters or locals of type 'RS' cannot be declared in async methods or async lambda expressions.
```

Comment:
>// (8,16): error CS4012: Parameters or locals of type 'RS' cannot be declared in async methods or async lambda expressions. [](http://example.com/codeflow?start=16&length=123)

We probably should add a dedicated error for his case, that doesn't talk about locals and parameters